src/librustdoc/html/static/js/search.js

```diff
@@ -127,13 +118,196 @@ window.initSearch = function(rawSearchIndex) {
+        function isStopCharacter(c) {
+            return isWhitespace(c) || "),>-=".indexOf(c) !== -1;
+        }
+        function getStringElem(query) {
+            query.pos += 1;
+            while (query.pos < query.length && query.val[query.pos] !== "\"") {
+                if (query.val[query.pos] === "\\") {
+                    // We ignore the next coming character.
+                    query.pos += 1;
+                }
+                query.pos += 1;
+            }
+            // To skip the quote at the end.
+            query.pos += 1;
+        }
+        function skipWhitespaces(query) {
+            var c;
+            while (query.pos < query.length) {
+                c = query.val[query.pos];
+                if (!isWhitespace(c)) {
+                    break;
+                }
+                query.pos += 1;
+            }
+
+        }
+        function skipStopCharacters(query) {
+            var c;
+            while (query.pos < query.length) {
+                c = query.val[query.pos];
+                if (!isStopCharacter(c)) {
+                    break;
+                }
+                query.pos += 1;
+            }
+        }
+        function isPathStart(query) {
+            var pos = query.pos;
+            return pos + 1 < query.length && query.val[pos] === ':' && query.val[pos + 1] === ':';
+        }
+        function isReturnArrow(query) {
+            var pos = query.pos;
+            return pos + 1 < query.length && query.val[pos] === '-' && query.val[pos + 1] === '>';
+        }
+        function removeEmptyStringsFromArray(x) {
+            for (var i = 0, len = x.length; i < len; ++i) {
+                if (x[i] === "") {
+                    x.splice(i, 1);
+                    i -= 1;
+                }
+            }
+        }
+        function createQueryElement(elems, val, generics, isExact) {
+            removeEmptyStringsFromArray(generics);
+            if (val === '*' || (val.length === 0 && generics.length === 0)) {
+                return;
+            }
+            var paths = val.split("::");
+            removeEmptyStringsFromArray(paths);
+            // In case we only have something like `<p>`, there is no name but it remains valid.
+            if (paths.length === 0) {
+                paths = [""];
+            }
+            elems.push({
+                isExact: isExact,
+                name: val,
+                fullPath: paths,
+                pathWithoutLast: paths.slice(0, paths.length - 1),
+                pathLast: paths[paths.length - 1],
+                generics: generics,
+            });
+        }
+        function getNextElem(query, elems) {
+            var isExact = false;
+            var generics = [];
+
+            skipStopCharacters(query);
+            var start = query.pos;
+            var end = start;
+            // We handle the strings on their own mostly to make code easier to follow.
+            if (query.val[query.pos] === "\"") {
```

Comment:
Since this is part of getNextElem, it looks like it's still trying to handle quotes in the middle of a query (e.g. `Vec<"usize">`). I thought we decided quotes would only be valid when surrounding a full query, with no special `<>()` in it.

Reply:
They're handled but ignored when we're matching against generics (all checks are string comparisons so "exact").